src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -865,11 +895,72 @@ def process_nw_flow_log_show_namespace(cmd, namespace):
 
 
 def process_nw_topology_namespace(cmd, namespace):
+    from msrestazure.tools import is_valid_resource_id, resource_id, parse_resource_id
+    SubResource = cmd.get_models('SubResource')
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
     location = namespace.location
+    rg = namespace.target_resource_group_name
+    vnet = namespace.target_vnet
+    subnet = namespace.target_subnet
+
+    vnet_id = vnet if is_valid_resource_id(vnet) else None
+    subnet_id = subnet if is_valid_resource_id(subnet) else None
+
+    if rg and not vnet and not subnet:
+        # targeting resource group - OK
+        pass
+    elif subnet:
+        subnet_usage = CLIError('usage error: --subnet ID | --subnet NAME --resource-group NAME --vnet NAME')
+        # targeting subnet - OK
+        if subnet_id and (vnet or rg):
+            raise subnet_usage
+        elif not subnet_id and (not rg or not vnet or vnet_id):
+            raise subnet_usage
+        if subnet_id:
+            rg = parse_resource_id(subnet_id)['resource_group']
+            namespace.target_subnet = SubResource(subnet)
+        else:
+            subnet_id = subnet_id or resource_id(
+                subscription=subscription_id,
+                resource_group=rg,
+                namespace='Microsoft.Network',
+                type='virtualNetworks',
+                name=vnet,
+                child_type_1='subnets',
+                child_name_1=subnet
+            )
+            namespace.target_resource_group_name = None
+            namespace.target_vnet = None
+            namespace.target_subnet = SubResource(subnet_id)
+    elif vnet:
+        # targeting vnet - OK
+        vnet_usage = CLIError('usage error: --vnet ID | --vnet NAME --resource-group NAME')
+        if vnet_id and (subnet or rg):
+            raise vnet_usage
+        elif not vnet_id and not rg or subnet:
```

Comment:
Based on the error message, should it be `not vnet_id and (not rg or not vnet)`?